### PR TITLE
Azure: Query resource managers for some of the high priority resource types: CosmoDB, IP Address,  Load Balancer, Key Vault, App Service

### PR DESCRIPTION
**Pull request by anastasiia-zolochevska**
Task - https://github.com/capitalone/cloud-custodian/issues/2253
Very basic query resource managers and basic tests for them.

### CODE REVIEW

**Comment by kapilt** [comment]
pretty cool and looks good, i don't want to loose the mapping of test to arm template that it depends on, any chance we could do a decorator, even if its just a dummy to start to record this mapping more explicitly, ie something like
```
@arm_template('xyz.json')
def test_foo(self):
```

File: tools/c7n_azure/c7n_azure/entry.py [hunk 1]
```diff
@@ -17,6 +17,11 @@
 import c7n_azure.resources.storage
 import c7n_azure.resources.resourcegroup
 import c7n_azure.resources.sqlserver
+import c7n_azure.resources.load_balancer
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/entry.py]: (minor thing) Maybe we can start sorting this alphabetically. It keeps it nice and organized.

**Comment by stefangordon** [comment]
I think we if can swap out to the decorators on the tests and Andy's feedback in an update then this is good to go.

File: tools/c7n_azure/tests/test_load_balancer.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function, unicode_literals
+from azure_common import BaseTest, arm_template
+
+
+# Recorded using template: storage
```
Comment (stefangordon) [inline tools/c7n_azure/tests/test_load_balancer.py]: I think left over comment to remove

File: tools/c7n_azure/tests/test_public_ip.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function, unicode_literals
+from azure_common import BaseTest, arm_template
+
+
+# Recorded using template: storage
```
Comment (stefangordon) [inline tools/c7n_azure/tests/test_public_ip.py]: another comment

**Pull request merged**